src/test/java/kostka/moviecatalog/service/integrationTests/StatisticServiceIntegrationTest.java

```diff
@@ -0,0 +1,82 @@
+package kostka.moviecatalog.service.integrationTests;
+
+import kostka.moviecatalog.MovieCatalogApplication;
+import kostka.moviecatalog.entity.counters.AtomicCounter;
+import kostka.moviecatalog.entity.counters.Counter;
+import kostka.moviecatalog.entity.counters.SynchronizedCounter;
+import kostka.moviecatalog.service.StatisticService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = MovieCatalogApplication.class)
+@Transactional
+public class StatisticServiceIntegrationTest {
+
+    public static final int THREADS = 3;
+    public static final int FROM_RANGE = 0;
+    public static final int TO_RANGE = 1000;
+
+    @Autowired
+    private StatisticService statisticService;
+
+    /***
+     * Test verifies that not synchronized counter is not incremented correctly
+     * sometimes it could happen that the test fail, but it is not probable
+     * @throws InterruptedException .
+     */
+    @Test
+    public void multiThreadNonSyncCounterIntegrationTest() throws InterruptedException {
+        ExecutorService service = Executors.newFixedThreadPool(THREADS);
+        Counter counter = new Counter();
+
+        IntStream.range(FROM_RANGE, TO_RANGE)
```

Comment:
Zajimalo by me, tohle ti pada tak nejak konzistentne? (soudim podle komentu)

Priznam se, ze si nejsem jisty, jestli vytovreni velkeho mnozstvi nezávislých a krátkých jobů ti zajistí s dobrou pravděpodobností, že se to někde sejde tak, aby ta inkrementace selhala.

Reply:
V asertu mám, že Result not equals to... Takze je vidět že neosetreny counter skoro nikdy nemá výsledek 1000

Reply:
Zatím test vždycky prošel, ale nevím jaká je úspěšnost že test spadne